Scoop/Controller/Ride Posting/InitialPostRideViewController.swift

```diff
@@ -0,0 +1,294 @@
+        taxiButton.setTitleColor(.black, for: .normal)
+        taxiButton.titleLabel?.font = UIFont(name: "SFPro", size: 16)
+        taxiButton.titleLabel?.adjustsFontSizeToFitWidth = true
+        taxiButton.setImage(UIImage(systemName: "circle"), for: .normal)
+        taxiButton.setImage(UIImage(systemName: "circle.inset.filled"), for: .selected)
+        taxiButton.tintColor = .black
+        taxiButton.titleEdgeInsets = UIEdgeInsets(top: 0, left: 10, bottom: 0, right: 0)
+        taxiButton.imageEdgeInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 10)
+        taxiButton.addTarget(self, action: #selector(taxiButtonToggle), for: .touchUpInside)
+        view.addSubview(taxiButton)
+        
+        taxiButton.snp.makeConstraints { make in
+            make.top.equalTo(studentDriverButton.snp.bottom).inset(-20)
+            make.leading.equalTo(studentDriverButton.snp.leading)
+            make.width.equalTo(180)
+        }
+    }
+    
+    private func setupDepartureTextField() {
+        let labelFont = UIFont(name: "Rambla-Regular", size: 16)
+        let textFieldBorderWidth = 1.0
+        let textFieldCornerRadius = 8.0
+        let textFieldFont = UIFont(name: "SFPro", size: 16)
+        
+        let departureLabel = UILabel()
+        departureLabel.text = "Departure"
+        departureLabel.font = labelFont
+        departureLabel.accessibilityLabel = "name"
+        departureLabel.textColor = .offBlack
+        
+        departureTextField.layer.borderWidth = textFieldBorderWidth
+        departureTextField.layer.borderColor = UIColor.textFieldBorderColor.cgColor
+        departureTextField.layer.cornerRadius = textFieldCornerRadius
+        departureTextField.font = textFieldFont
+        departureTextField.textColor = .darkGray
+        departureTextField.attributedPlaceholder = NSAttributedString(string:"Departure location", attributes: [NSAttributedString.Key.foregroundColor: UIColor.offBlack])
+        departureTextField.addTarget(self, action: #selector(presentDepartureSearch), for: .touchDown)
+        
+        // Sets up the icon inside the text field.
+        let iconContainer = UIView(frame: CGRect(x: 0, y: 0, width: 25, height: 15))
+        let departureIcon = UIImageView(frame: CGRect(x: 10, y: -2.5, width: 20, height: 20))
+        departureIcon.image = UIImage(named: "locationIcon")
+        departureIcon.contentMode = .scaleAspectFit
+        iconContainer.addSubview(departureIcon)
+        
+        departureTextField.leftView = iconContainer;
+        departureTextField.leftViewMode = UITextField.ViewMode.always
+        departureTextField.leftViewMode = .always
+        view.addSubview(departureTextField)
+        
+        departureTextField.snp.makeConstraints { make in
+            make.leading.equalTo(titleLabel.snp.leading)
+            make.top.equalTo(taxiButton.snp.bottom).inset(-24)
+            make.trailing.equalToSuperview().inset(32)
```

Comment:
As a side note - I like to inset relative to the superview (because I know this view is contained inside the superview), but when I am positioning views relative to each other, I like to use offset